### ISSUE TITLE
Aleth fails on hive: missing `difficulty` on `eth_getBlockByNumber` for block `0`

### ISSUE THREAD

**Issue by holiman** [ethereum/aleth#5444]
Aleth currently fails on hive, because `aleth` does not include the  `difficulty` when answering to an `eth_getBlockByNumber`request for block `0`. This causes the `ethclient` used by the simulator to throw an internal error.

Request
``` 
POST / HTTP/1.1
Host: 172.17.0.23:8545
User-Agent: Go-http-client/1.1
Content-Length: 78
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{"jsonrpc":"2.0","id":1,"method":"eth_getBlockByNumber","params":["0x0",true]}

```
Response: 
```
Server: BaseHTTP/0.6 Python/3.6.6
Date: Mon, 21 Jan 2019 09:14:54 GMT
Content-type: application/json
Content-length: 1326
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: POST, GET, OPTIONS
Access-Control-Allow-Headers: content-type


{"id":1,"jsonrpc":"2.0","result":{"author":"0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba","extraData":"0x42","gasLimit":"0xf4240","gasUsed":"0x0","hash":"0xb2c9d6c3efe05eeb7cdd03955d76775c966b5818e316d4b43b2d1e52033dce1c","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","miner":"0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba","number":"0x0","parentHash":"0x0000000000000000000000000000000000000000000000000000000000000000","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","size":"0x0","stateRoot":"0x3209135bff9bc802c3336207ad37d3610f48625a321a5583cc1b6267e5be8d50","timestamp":"0x3b6","totalDifficulty":"0x20000","transactions":[],"transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","uncles":[]}}

```
data: 
```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "result": {
    "author": "0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba",
    "extraData": "0x42",
    "gasLimit": "0xf4240",
    "gasUsed": "0x0",
    "hash": "0xb2c9d6c3efe05eeb7cdd03955d76775c966b5818e316d4b43b2d1e52033dce1c",
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "miner": "0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba",
    "number": "0x0",
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "receiptsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
    "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "size": "0x0",
    "stateRoot": "0x3209135bff9bc802c3336207ad37d3610f48625a321a5583cc1b6267e5be8d50",
    "timestamp": "0x3b6",
    "totalDifficulty": "0x20000",
    "transactions": [],
    "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
    "uncles": []
  }
}
```

**Comment by gumb0** [comment]
@winsvega I see from the code it's not returned for the config with `NoProof` seal engine. SealEngine needs `jsInfo` method implementation, similar to Ethash
https://github.com/ethereum/aleth/blob/08906ebc27283d2136d66cdabff9bb380c04c83b/libethashseal/Ethash.cpp#L79-L82

**Comment by gumb0** [comment]
@holiman Are `nonce`, `seedHash`, `mixHash`, `boundary` of any importance? The don't make much sense for NoProof and we don't return them either

**Comment by holiman** [comment]
TLDR; Yep, should be fine

I think we can leave out `mixHash` and `nonce` (not `required`), and from what I can tell, we don't have the concept of `seedHash` or `boundary`, so those would also just be ignored. 
https://github.com/ethereum/go-ethereum/blob/master/core/types/block.go#L70 : 
```
type Header struct {
	ParentHash  common.Hash    `json:"parentHash"       gencodec:"required"`
	UncleHash   common.Hash    `json:"sha3Uncles"       gencodec:"required"`
	Coinbase    common.Address `json:"miner"            gencodec:"required"`
	Root        common.Hash    `json:"stateRoot"        gencodec:"required"`
	TxHash      common.Hash    `json:"transactionsRoot" gencodec:"required"`
	ReceiptHash common.Hash    `json:"receiptsRoot"     gencodec:"required"`
	Bloom       Bloom          `json:"logsBloom"        gencodec:"required"`
	Difficulty  *big.Int       `json:"difficulty"       gencodec:"required"`
	Number      *big.Int       `json:"number"           gencodec:"required"`
	GasLimit    uint64         `json:"gasLimit"         gencodec:"required"`
	GasUsed     uint64         `json:"gasUsed"          gencodec:"required"`
	Time        *big.Int       `json:"timestamp"        gencodec:"required"`
	Extra       []byte         `json:"extraData"        gencodec:"required"`
	MixDigest   common.Hash    `json:"mixHash"`
	Nonce       BlockNonce     `json:"nonce"`
}
```
